src/unity/python/turicreate/toolkits/drawing_recognition/drawing_recognition.py

```diff
@@ -0,0 +1,480 @@
+            self.fc1 = _nn.Dense(units=128, activation='relu')
+            self.fc2 = _nn.Dense(units=num_classes, activation=None)
+
+    def hybrid_forward(self, F, x):
+        x = self.pool1(self.conv1(x))
+        x = self.pool2(self.conv2(x))
+        x = self.pool3(self.conv3(x))
+        x = self.flatten(x)
+        x = self.fc1(x)
+        x = self.fc2(x)
+        return F.softmax(x)
+
+def create(input_dataset, annotations=None, num_epochs=100, feature="bitmap", model=None,
+           classes=None, batch_size=256, max_iterations=0, verbose=True, 
+           **kwargs):
+    """
+    Create a :class:`DrawingRecognition` model.
+    """
+    start_time = _time.time()
+
+    is_stroke_input = (input_dataset[feature].dtype != _tc.Image)
+
+    if is_stroke_input:
+        # This only works on macOS right now
+        dataset = _extensions._drawing_recognition_prepare_data(
+            input_dataset, "bitmap", "label")
+    else:
+        dataset = input_dataset
+
+    column_names = ['Iteration', 'Loss', 'Elapsed Time']
+    num_columns = len(column_names)
+    column_width = max(map(lambda x: len(x), column_names)) + 2
+    hr = '+' + '+'.join(['-' * column_width] * num_columns) + '+'
+
+    progress = {'smoothed_loss': None, 'last_time': 0}
+    iteration = 0
+
+    if classes is None:
+        classes = dataset['label'].unique()
+    classes = sorted(classes)
+    class_to_index = {name: index for index, name in enumerate(classes)}
+
+    def update_progress(cur_loss, iteration):
+        iteration_base1 = iteration + 1
+        if progress['smoothed_loss'] is None:
+            progress['smoothed_loss'] = cur_loss
+        else:
+            progress['smoothed_loss'] = (0.9 * progress['smoothed_loss'] 
+                + 0.1 * cur_loss)
+        cur_time = _time.time()
+
+        # Printing of table header is deferred, so that start-of-training
+        # warnings appear above the table
+        if verbose and iteration == 0:
+            # Print progress table header
+            print(hr)
+            print(('| {:<{width}}' * num_columns + '|').format(*column_names, 
+                width=column_width-1))
+            print(hr)
+
+        if verbose and (cur_time > progress['last_time'] + 10 or
+                        iteration_base1 == max_iterations):
+            # Print progress table row
+            elapsed_time = cur_time - start_time
+            print(
+                "| {cur_iter:<{width}}| {loss:<{width}.3f}| {time:<{width}.1f}|".format(
+                cur_iter=iteration_base1, loss=progress['smoothed_loss'],
+                time=elapsed_time , width=column_width-1))
+            progress['last_time'] = cur_time
+
+    loader = _SFrameRecognitionIter(dataset, batch_size,
+                 feature_column='bitmap',
+                 target_column='label',
+                 class_to_index=class_to_index,
+                 load_labels=True,
+                 shuffle=True,
+                 io_thread_buffer_size=0,
```

Comment:
If we're not actually using an IO thread, let's just not include the relevant code in the iterator implementation. If you decide we need it, you can copy it then. Hopefully will have a better, shared solution before then